gestalt-module/src/main/java/org/terasology/gestalt/naming/Name.java

```diff
@@ -39,39 +39,48 @@
      */
     public static final Name EMPTY = new Name("");
 
+    private final String normalizedName;
     private final String originalName;
-    private final String normalisedName;
 
     public Name(String name) {
         Preconditions.checkNotNull(name);
         this.originalName = name;
-        this.normalisedName = name.toLowerCase(Locale.ENGLISH);
+        this.normalizedName = name.toLowerCase(Locale.ENGLISH);
+    }
+
+    /**
+     * @return The original <b>case-sensitive</b>  name when string is passed into name
+     */
+    public String displayName() {
+        return originalName;
     }
 
     /**
      * @return Whether this name is empty (equivalent to an empty string)
      */
     public boolean isEmpty() {
-        return normalisedName.isEmpty();
+        return normalizedName.isEmpty();
     }
 
     /**
      * @return The Name in lowercase consistent with Name equality (so two names that are equal will have the same lowercase)
+     * @deprecated
      */
     public String toLowerCase() {
-        return normalisedName;
+        return normalizedName;
     }
 
     /**
      * @return The Name in uppercase consistent with Name equality (so two names that are equal will have the same uppercase)
+     * @deprecated
      */
     public String toUpperCase() {
```

Comment:
Same here for this deprecation.

```suggestion
     * @deprecated This is scheduled for removal in upcoming versions. 
     *             Use {@code toString} or {@code displayName} instead.
     *             Note that a Name should not be transformed to a String for further processing.
     */
    @Deprecated
    public String toUpperCase() {
```